axlearn/cloud/gcp/job.py

```diff
@@ -508,8 +516,9 @@ def _build_pod(self) -> Nested[Any]:
             logging.info("Found tier=%s in env. Using reservation=%s", tier, cfg.reservation)
             selector.update({"cloud.google.com/reservation-name": cfg.reservation})
         else:
```

Comment:
What about using `tier == "1"` for spot and `tier == "2"` for on-demand? You can configure the quota file s.t. there's only on-demand quota like
```
[[total_resources]]
v5litepod = 0

[[total_resources]]
v5litepod = 0

[[total_resources]]
v5litepod = X

...
```

Which corresponds to 0 quota for tier 0/1 and X quota on tier 2. More generally we can also make the logic for `tier -> selector` mapping configurable, for more flexible schemes. WDYT?